code-challenges.test.js

```diff
@@ -18,19 +20,91 @@ const fibonacciLength1 = 6
 const fibonacciLength2 = 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+test('fibonacci sequence for fibonacciLength1', () => {
+    const result = generateFibonacci(fibonacciLength1)
+    expect(result).toEqual([1, 1, 2, 3, 5, 8])
+})
+
+test('fibonacci sequence for fibonacciLength2', () => {
+    const result = generateFibonacci(fibonacciLength2)
+    expect(result).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+})
+
+
+
 // b) Create the function that makes the test pass.
+const generateFibonacci = (length) => {
+    if (length <= 0) {
+        return []
+    } else if (length === 1) {
+        return [1]
+    } else if (length === 2) {
+        return [1, 1]
+    }
 
-// Pseudo code:
+
+const fibonacciArray = [1, 1]
+for (let i = 2; i < length; i++) {
+    const nextFibonacci = fibonacciArray[i - 1] + fibonacciArray[i - 2]
+    fibonacciArray.push(nextFibonacci)
+}
+
+return fibonacciArray
+}
+
+const result1 = generateFibonacci(fibonacciLength1)
+const result2 = generateFibonacci(fibonacciLength2)
+
+console.log(result1)
+console.log(result2)
```

Comment:
The jest test will perform the function call. These lines are not needed outside the test.